lib/documents.js

```diff
@@ -2472,6 +2472,95 @@ function writeDocs(jobState, val, batchdef, writeBatchArray, writerId){
       });
 }
 
+
+Documents.prototype.queryAll = function queryAllDocuments(query, options) {
+  return queryAllDocumentsImpl.call(
+      this, query, options
+  );
+};
+
+function queryAllDocumentsImpl(query, jobOptions) {
+   if(!query){
+     throw new Error('Query cannot be null or undefined.');
+   }
+
+  let path = '/v1/internal/forestinfo';
+  let connectionParams = this.client.getConnectionParams();
+  let requestOptions = mlutil.copyProperties(connectionParams);
+
+
+  requestOptions.method = 'GET';
+  requestOptions.headers = {
+    'Accept':       'application/json',
+  };
+
+  requestOptions.path = mlutil.databaseParam(connectionParams, path);
+  let operation = new Operation(
+      'read forestInfo', this.client, requestOptions, 'empty', 'single'
+  );
+
+  let jobState = {
+    docInstance : this,
+    requesterCount: 0,
+    jobOptions: (jobOptions)? mlutil.copyProperties(jobOptions):{}
+  };
+
+  // Check for jobOptions.batchSize
+  jobState.jobOptions.batchSize = (jobState.jobOptions.batchSize && jobState.jobOptions.batchSize >0) ?
+      jobState.jobOptions.batchSize:1000;
+
+  jobState.stream = (jobState.jobOptions.batchSize === 1)? new stream.PassThrough():
+      new stream.PassThrough({objectMode: true});
+  jobState.query = query;
+  requester.startRequest(operation).result(mlutil.callbackOn(jobState, onQueryUrisRequest))
+      .catch((err) => operation.stream.emit('error', err));
+
+  return jobState.stream;
+}
+
+function onQueryUrisRequest(output) {
+
+  const jobState = this;
+  let endpoint = '/v1/internal/uris?format=json';
+  let wrapper = qb.makeSearchBody(jobState.query);
+
+  let categories  = wrapper.categories;
+  let optionsName = wrapper.optionsName;
+  let pageStart   = wrapper.pageStart;
+  let pageLength  = wrapper.pageLength;
+  let txid        = mlutil.convertTransaction(wrapper.txid);
+  let transform   = wrapper.transform;
+  let view        = wrapper.view;
+  let searchBody  = wrapper.searchBody;
+
+  let connectionParams = this.client.getConnectionParams();
+  let requestOptions = mlutil.copyProperties(connectionParams);
+
+  requestOptions.method = 'GET';
+  requestOptions.headers = {
+    'Accept':       'application/json',
+  };
+
+  requestOptions.path = mlutil.databaseParam(connectionParams, endpoint);
+  let operation = new Operation(
+      'read uris', this.client, requestOptions, 'empty', 'single'
+  );
+
+  operation.requestBody = searchBody;
+  operation.validStatusCodes  = [200, 204, 404];
+  operation.inlineAsDocument  = false;
+
+  requester.startRequest(operation).result(jobState, onQueryAllDocs)
```

Comment:
I believe this line would need to execute the call() method on the function reference to set the this object, as in:

requester.startRequest(operation).result.call(jobState, onQueryAllDocs)